res/res_rtp_asterisk.c

```diff
@@ -4257,12 +4259,31 @@ static int ast_rtp_dtmf_begin(struct ast_rtp_instance *instance, char digit)
 		return -1;
 	}
 
-	/* Grab the payload that they expect the RFC2833 packet to be received in */
-	payload = ast_rtp_codecs_payload_code_tx(ast_rtp_instance_get_codecs(instance), 0, NULL, AST_RTP_DTMF);
+	if (rtp->lasttxformat == ast_format_none) {
+		/* No audio frames have been written yet so we have to lookup both the preferred payload type and bitrate. */
+		payload_format = ast_rtp_codecs_get_preferred(ast_rtp_instance_get_codecs(instance));
+		if(payload_format) {
+			/* If we have a preferred type, use that. Otherwise default to 8K. */
+			sample_rate = ast_format_get_codec_rate(payload_format);
```

Comment:
This will result in the wrong sample rate for G.722, there is an ast_rtp_get_rate function which will return the sample rate of a format and handle the G.722 scenario